targets/actionscript/make.js

```diff
@@ -123,6 +123,55 @@ function getVerticalNameDefault() {
     return "null";
 }
 
+function getValidPlayFabActionScriptNamespacePrefix(datatype) {
+
+    var asIndividualNamespaces = datatype.classNameSpace.split('.');
+
+    var asNamespace = "com.";
+
+    if (asIndividualNamespaces.length < 2)
+    {
+        throw new Error("Error in Generating API Model: Namespaces are expected to be in the form playfab.[SDK].[Name]");
+    }
+
+    for (var i = 0; i < asIndividualNamespaces.length; i++) {
+        if (i < 1) {
+            asNamespace += asIndividualNamespaces[i].toLowerCase();
+        }
+        else {
+            asNamespace += asIndividualNamespaces[i];
+        }
+
+        // we expect the input to be playfab.[SDK].[Name]
+        // but the actual datatype-callable prefix namespace in ActionScript is com.playfab.[SDK][Name].
+        if (i < asIndividualNamespaces.length - 2) {
+            asNamespace += ".";
```

Comment:
String constructions like this in JS are very poor performance.
In JS, you should build a list of strings:
var collection = ["com.", "playfab", ".", "classnamespace", ".", "classname"]
and then do a join:
"".join(collection);

Reply:
Repeated += on a string in scripting languages is an N^2 operation.
The array plus join is much closer to just an N operation.